### PR TITLE
Vegan Trait

### PR DESCRIPTION
**Pull request by Holli-Git**
#### Summary
Content "Vegan Trait"

#### Purpose of change

I've always liked the challenge of going without animal products in game, but found it very difficult to impose restrictions myself, as items would often have animal products without me even knowing. So, this PR adds a vegan trait, along with some miscellaneous fixes such as human blood now being cannibalistic.  

#### Describe the solution

Adds a vegan trait, along with some vegan friendly options for the more important clothing items. 

#### Describe alternatives you've considered

Not doing this?

#### Testing

Compiled, tested, no errors.

### CODE REVIEW

**Comment by anoobindisguise** [comment]
You should add gutskin to the list of vegan non wearables. There are like 2 items in the game that have it and they don't show up naturally, but it is made from stomach.

**Comment by Paprika-H** [comment]
> You should add gutskin to the list of vegan non wearables. There are like 2 items in the game that have it and they don't show up naturally, but it is made from stomach. Make sure to add flag_URSINE_HONEY to the food blacklist!

Good catch!

File: src/character_attire.cpp [hunk 1]
```diff
@@ -26,7 +26,13 @@ static const flag_id json_flag_ONE_PER_LAYER( "ONE_PER_LAYER" );
 
 static const itype_id itype_shoulder_strap( "shoulder_strap" );
 
+static const material_id material_acidchitin( "acidchitin" );
+static const material_id material_chitin( "chitin" );
+static const material_id material_fur( "fur" );
+static const material_id material_gutskin( "gutskin" );
+static const material_id material_leather( "leather" );
 static const material_id material_wool( "wool" );
+static const material_id material_bone( "bone" );
```
Comment (dseguin) [inline src/character_attire.cpp:35]:
```suggestion
static const material_id material_acidchitin( "acidchitin" );
static const material_id material_bone( "bone" );
static const material_id material_chitin( "chitin" );
static const material_id material_fur( "fur" );
static const material_id material_gutskin( "gutskin" );
static const material_id material_leather( "leather" );
static const material_id material_wool( "wool" );
```

File: src/item_factory.cpp [hunk 2]
```diff
@@ -89,10 +89,13 @@ static const item_category_id item_category_weapons( "weapons" );
 static const item_group_id Item_spawn_data_EMPTY_GROUP( "EMPTY_GROUP" );
 
 static const material_id material_bean( "bean" );
+static const material_id material_blood( "blood" );
 static const material_id material_egg( "egg" );
 static const material_id material_flesh( "flesh" );
 static const material_id material_fruit( "fruit" );
 static const material_id material_garlic( "garlic" );
+static const material_id material_hblood( "hblood" );
+static const material_id material_bone( "bone" );
```
Comment (dseguin) [inline src/item_factory.cpp:98]:
```suggestion
static const material_id material_blood( "blood" );
static const material_id material_bone( "bone" );
static const material_id material_egg( "egg" );
static const material_id material_flesh( "flesh" );
static const material_id material_fruit( "fruit" );
static const material_id material_garlic( "garlic" );
static const material_id material_hblood( "hblood" );
```

File: src/consumption.cpp [hunk 3]
```diff
@@ -825,6 +834,11 @@ ret_val<edible_rating> Character::can_eat( const item &food ) const
         return ret_val<edible_rating>::make_failure( INEDIBLE_MUTATION,
                 _( "The thought of eating that makes you feel sick." ) );
     }
```
Comment (dseguin) [inline src/consumption.cpp:836]:
```suggestion
    }
    const std::array<flag_id, 5> vegan_blacklist {{
        json_flag_ALLERGEN_MEAT, json_flag_ALLERGEN_EGG,
        json_flag_ALLERGEN_MILK, json_flag_ANIMAL_PRODUCT,
        flag_URSINE_HONEY
    }};
```

File: src/consumption.cpp [hunk 4]
```diff
@@ -158,6 +161,12 @@ static const std::array<flag_id, 2> herbivore_blacklist {{
         json_flag_ALLERGEN_MEAT, json_flag_ALLERGEN_EGG
     }};
 
+static const std::array<flag_id, 5> vegan_blacklist {{
+        json_flag_ALLERGEN_MEAT, json_flag_ALLERGEN_EGG,
+        json_flag_ALLERGEN_MILK, json_flag_ANIMAL_PRODUCT,
+        flag_URSINE_HONEY
+    }};
+
```
Comment (dseguin) [inline src/consumption.cpp:169]:
```suggestion
```

**Pull request merged**